### ISSUE TITLE
Foggy Island

### ISSUE THREAD

**Issue by Draylar** [fabric-community/spooky-time#114]
An island located somewhere far out in the deep ocean. The location isn't mapped, as everyone who has set out to do so never returned.

  Introducing Foggy Island (name TBD); an endgame location with some of the more difficult content in the mod. The Foggy Island is not visible until you enter, and escaping is difficult: thick fog clouds your vision, and you're seemingly drawn in closer and closer despite your attempts to turn back...

-------------------------------------------------------------------

_You hear the sound of crumbling paper and start to smell the sea._
    It's been a while since you started traveling in your boat. A strange light caught your eye off the coast, and you couldn't help but investigate. Despite the fact that the bay itself was fairly small, it had been a good 30 minutes since you had left-- _much_ longer than it should take on a normal day. 
    The fog was getting mysteriously thick. It felt like you had water stuck in your nose; your feet had been soaked long ago from the rising puddle in your boat. While you had a feeling continuing onwards would be dangerous, you still sought out that pail light, which would occasionally spark in the distant fog, guiding your path.
    Your ability to see more than a few feet had disappeared a few minutes ago; and if you ever managed to open your eyes against the pelting rain which had started at a similar time, your vision was met with pure gray and a single light, still years in front of you.
   
_Thump._
    A rock? Something hard brushed up against the boat, and the rain seemed to have stopped momentarily... almost as if something large was looming ahead...
        like a large, pillaring rock. Despite your almost non-existant sight, you could barely make out a large spike protruding from the crashing surface of the water. Next to it sat several other spikes of similar spikes, all reaching for the skies. 
    Your thoughts of climbing up the towers, despite the rain, were cut short as a screech emanated from the sky. Through the fog, which had started to lessen up at this point, flew several large shadows. Discovering the source of the shadows was not on your to-do list, and would most likely stay off it for quite a long time.

_Tsshh..._    
    The light, which had been dashing between the pillar's gaps for quite some time now, suddenly came to a rest in your vision, just as it had started. It took you a second to notice, but the churning of the boat in the waves suddenly stopped: as a matter of fact, the _boat_ was stopped. Moving your hands to the side of the boat, you feel around for anything of interest. A brittle, soft texture responds back... grainy to the touch, yet packed together. Sand?
    You had hit land. You step out of the boat and rub your eyes to push the water aside; your vision gradually returns and you take a look at what you had stumbled upon.
    To say it was the king of the previous pillars was an understatement. If the stone spikes reached for the skies, this skewer reached for the stars. It wouldn't be a bad estimate to say the fog in the sky reached around 1/3 way up the peak, but you couldn't confirm without getting closer. 
    An opportunity to go closer appears just as you wish for it-- the light, which had been stationary till this point, started to slowly ascend up the wall of stone. In the wake of the luminance, you can make out a small, steep path circling up the mountain.

    Will you follow it?

    In every slate of darkness...
       is a small pillar of light.

--------------------------------------------------------------------

**Meta details on entering:**
    When you enter the fog of the _Foggy Island Zone_, you're inside a custom biome with thick fog. If you choose to continue to travel in (which ends up being around 100 blocks), your screen will start to turn gray from the fog, until you can _only_ see a solid gray. At this point, your dimension is swapped, but the dimension changing screen is replaced with pure gray as well-- in other words, an attempt at seamless dimension transportation. This mechanic exists to allow more space in the fog; the zone in the spooky dimension is only around 200x200 big, but the zone itself is in the upper zones of 750x750. Turning backward while in the dimension results in heading back to the spooky dimension with the same mechanic. The player is lead by a glowing light in the fog, which leads to the island.

**Non-narrative description:**
  The _Foggy Island_ consists of 4 parts: inner island, outer island, outer spikes, and foggy surface. You'll first enter the foggy surface, which is an area with almost no visibility and nothing but water. Sea beasts lurk in the depth: some are waiting for you to take a dip, while others are more than happy to rock your boat. Pirate ships are a great way to travel through!

  Once you're close enough in, you'll start to see spikes jutting out of the water. Large shadows lurk at the base of the spikes (they are hungry), and soaring ghosts will take you out of the air if you attempt to climb up a spike to see further out.

  If you make it past the spikes, you'll hit a length of open space, and then hit the beach of the island. The fog at this point is less severe than earlier, but it still prevents visibility far up and limits you to around 50 blocks of view horizontally.

  To make it up the mountain, you can take one of the several paths provided up the side. Pillaring is prohibited, and flying gets you struck down by lightning. Both have warning messages associated with the action: "_You start to regret pillaring up..._", and "_The smell of electricity fills the air... it may be best to land._"

  I don't have any concrete ideas as to what is at the top (or what the light is) just yet. I just want a cool area with fog and fun, seamless, instanced dimensional teleportation! Maybe the light could be the old ruler of the spooky world (flashback time ???) and the boss could be a giant stone-spike golem who climbs around the main mountain... :)

**Comment by valoeghese** [comment]
I reckon foggy island could be done via having a rare deep ocean variant of the sea biome group, which has an edge layer that replaces surrounding biome groups with the sea biome group, making it surrounded much by sea. This would only generate 6 or more large-biome-fractal-tiles out (manhattan distance)

**Comment by valoeghese** [comment]
On around Thursday I’ll start working on that if no one else has already

**Comment by Shrimpstress** [comment]
Foggy Island will be it’s own dimension, as mentioned in the original issue.

**Comment by Draylar** [comment]
Yes, the actual island would be a separate dimension. What we _do_ need is a way to calculate how far from the center the player is to scale fog amount (center = max fog); in other words, it should _basically_ be a circle biome with no different characteristics from the biomes around it (ignoring fog).
